src/middlewared/middlewared/plugins/interface/netif_linux/interface.py

```diff
@@ -41,55 +47,45 @@ def description(self, value):
 
     @property
     def mtu(self):
-        return self._read("mtu", int)
+        return self._mtu
 
     @mtu.setter
-    def mtu(self, mtu):
-        up = InterfaceFlags.UP in self.flags
-        run(["ip", "link", "set", "dev", self.name, "mtu", str(mtu)])
-        if up:
-            self.down()
-            self.up()
+    def mtu(self, value):
+        with NDB(log='off') as ndb:
+            with ndb.interfaces[self.orig_name] as dev:
+                dev['mtu'] = value
+                if dev['state'] == 'up':
+                    dev['state'] = 'down'
+                    dev['state'] = 'up'
```

Comment:
is line 59 superfluous ? I mean, you got here **because** it was already `up`, no ?